### PR TITLE
Fix noneditable mypy

### PR DESCRIPTION
**Pull request by vincentpierre**
## Motivation and Context

Installing in non-editable fails, adding `__init__.py` files fixes it according to @1heart 
Fixing mypy to allow for the tests to pass

## How Has This Been Tested

Added unit test which ensures habitat is importable when installing in non-editable mode.

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by 1heart — APPROVED** [review]
Thanks!

File: habitat/tasks/rearrange/multi_task/composite_task.py [hunk 1]
```diff
@@ -6,10 +6,10 @@
 
 import copy
 import os.path as osp
-from typing import Any, Dict, List, Optional
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
 
 import numpy as np
-import yaml
+import yaml  # type: ignore[import]
```
Comment (ASzot) [inline habitat/tasks/rearrange/multi_task/composite_task.py:12]: Good short term solution, but dependency on yaml is completely removed in this PR: https://github.com/facebookresearch/habitat-lab/pull/883

File: habitat/tasks/rearrange/multi_task/rearrange_pddl.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
     from habitat.datasets.rearrange.rearrange_dataset import RearrangeDatasetV0
 
 
-def parse_func(x: str) -> Tuple[str, List[str]]:
+def parse_func(x: str) -> Tuple[str, str]:
```
Comment (ASzot) [inline habitat/tasks/rearrange/multi_task/rearrange_pddl.py:40]: I also fixed this in this PR: https://github.com/facebookresearch/habitat-lab/pull/883

**Review by ASzot — APPROVED** [review]
Agreed the `__init__.py` file is important. Overall, [this PR](https://github.com/facebookresearch/habitat-lab/pull/883) addresses many of the typing / other issues in the rearrange part of the code that were also addressed in this PR.

But LGTM.

**Review by Skylion007 — APPROVED** [review]

**Comment by Skylion007** [comment]
I've been meaning to extract this out the stalled PR to make habitat-lab fully pip installable. Looks good to me. 

Habitat-Sim is already pip installable and even has its own unit test.

**Comment by vincentpierre** [comment]
Ok awesome. When I get the tests working I will merge it then. Thanks for the help.

**Pull request merged**